homeassistant/components/swiss_public_transport/sensor.py

```diff
@@ -6,45 +6,60 @@
 
 from opendata_transport import OpendataTransport
 from opendata_transport.exceptions import OpendataTransportError
-import voluptuous as vol
 
-from homeassistant.components.sensor import PLATFORM_SCHEMA, SensorEntity
+from homeassistant import config_entries, core
+from homeassistant.components.sensor import SensorEntity
 from homeassistant.const import CONF_NAME
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.aiohttp_client import async_get_clientsession
-import homeassistant.helpers.config_validation as cv
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
 import homeassistant.util.dt as dt_util
 
+from .const import (
+    ATTR_DELAY,
+    ATTR_DEPARTURE_TIME1,
+    ATTR_DEPARTURE_TIME2,
+    ATTR_DURATION,
+    ATTR_PLATFORM,
+    ATTR_REMAINING_TIME,
+    ATTR_START,
+    ATTR_TARGET,
+    ATTR_TRAIN_NUMBER,
+    ATTR_TRANSFERS,
+    CONF_DESTINATION,
+    CONF_START,
+    DOMAIN,
+)
+
 _LOGGER = logging.getLogger(__name__)
 
-ATTR_DEPARTURE_TIME1 = "next_departure"
-ATTR_DEPARTURE_TIME2 = "next_on_departure"
-ATTR_DURATION = "duration"
-ATTR_PLATFORM = "platform"
-ATTR_REMAINING_TIME = "remaining_time"
-ATTR_START = "start"
-ATTR_TARGET = "destination"
-ATTR_TRAIN_NUMBER = "train_number"
-ATTR_TRANSFERS = "transfers"
-ATTR_DELAY = "delay"
+SCAN_INTERVAL = timedelta(seconds=90)
 
-CONF_DESTINATION = "to"
-CONF_START = "from"
 
-DEFAULT_NAME = "Next Departure"
+async def async_setup_entry(
+    hass: core.HomeAssistant,
+    config_entry: config_entries.ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the sensor from a config entry created in the integrations UI."""
+    config = hass.data[DOMAIN][config_entry.entry_id]
 
+    sensor = await swiss_public_transport_setup(hass, config)
 
-SCAN_INTERVAL = timedelta(seconds=90)
+    try:
+        await sensor.async_update()
+    except OpendataTransportError:
+        _LOGGER.error(
+            "Check at http://transport.opendata.ch/examples/stationboard.html "
+            "if your station names are valid"
+        )
+        return
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Required(CONF_DESTINATION): cv.string,
-        vol.Required(CONF_START): cv.string,
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-    }
-)
+    async_add_entities(
+        [sensor],
+        update_before_add=True,
+    )
 
 
 async def async_setup_platform(
```

Comment:
So the thing is, we should not use YAML anymore. Instead we want to import the YAML using an import flow which creates a config entry for the YAML config.

Reply:
@joostlek Do you mean, it should not be possible anymore to configure via `configuration.yaml`?

Reply:
Ohh, that is sad...
But removing it will be a breaking change and I would rather not do that.

But I don't think this is up to me to decide. What you would say @fabaff ?

Reply:
Well, we have an import flow for that. You can then import the YAML so it will create an entry for you

Reply:
Ok, have not yet looked at that.

Reply:
@joostlek Did you refer to this when you were talking about `import flow`?
https://github.com/home-assistant/example-custom-config/blob/master/custom_components/example_load_platform/README.md

Reply:
Nope: checkout https://github.com/home-assistant/core/pull/104730/files#diff-3f88feab5a41e23e0057f3117af8a8effc16d9a136c2dbb07e293286a9acb0ab

(This is just to show how an import would work. I changed this code up a bit as its now a bit cleaner, so please checkout the version of suez water on dev)

Reply:
Thanks for this hint. I have a new proposition for it based on the Suez example.